File: AdminServer/AdminServerBackend/src/main/java/net/rickiekarp/admin/dao/UserDAO.java
//package net.rickiekarp.admin.dao;
//
//import java.util.List;
//
//import net.rickiekarp.admin.exception.DuplicateUserException;
//import net.rickiekarp.admin.exception.UserNotFoundException;
//import net.rickiekarp.admin.model.User;
//
//public interface UserDAO {
//
//    public void addUser(User user) throws DuplicateUserException;
//
//    public User getUser(int userId) throws UserNotFoundException;
//
//    public User getUser(String username) throws UserNotFoundException;
//
//    public void updateUser(User user) throws UserNotFoundException, DuplicateUserException;
//
//    public void deleteUser(int userId) throws UserNotFoundException;
//
//    public List<User> getUsers();
//
//}

File: LoginServer/src/main/java/net/rickiekarp/loginserver/dao/UserDAO.kt
package net.rickiekarp.loginserver.dao

import net.rickiekarp.foundation.model.Credentials
import net.rickiekarp.foundation.model.User

interface UserDAO {
    fun getUserFromToken(token: String): User?
    fun getUserByName(username: String): User?
    fun updateUserToken(user: User, token: String)
    fun registerUser(user: Credentials): User?
}
